cookbooks/multicomponent_steinberger/doc/multicomponent_steinberger.md

```diff
@@ -0,0 +1,159 @@
+instead of `Model names` and specifying `List of model operators = add`, we
+are able to add the function expression and the boundary temperature model
+expression to the adiabatic temperature profile.
+
+We use the half-space cooling model to set up a 100 Ma top boundary layer and
+a 3.5 Ga bottom boundary layer. We use `Function` to prescribe a circular
+thermal anomaly at the same position with the same size as the compositional
+anomaly. The temperature of theanomaly is the maximum value between 400 K and
+the temperature difference between the actual temperature profile and the ambient
+mantle adiabat. This setup would produce a thermal anomaly with an excess
+temperature of 400 K that deviates from the bottom thermal boundary layer smoothly
+({numref}`fig:temperature.multicompstn`:left).
+```{literalinclude} temperature.setup.prm
+```
+Since we use a set of arbitrary heat capacity, density, and thermal conductivity to
+calculate the thermal diffusivity (function constant `kappa`) and the heat capacity
+and density are obtained from the look-up table, you should choose the values that
+are representative for your model. You can do it by first setting `End time = 1`
+and running the model. You then find the representative heat capacity and density
+values for your model from the visualization output, and the thermal conductivity is
+set to be 1.5 K/m in the `material model`. Other than `kappa`, the other constants
+used in the `Function`, the adiabatic bottom temperature (`Tab`)and the age of the
+bottom thermal boundary layer (`age`), should also consistent with the adiabatic
+temperature setup of the model.
+
+You may also notice that there is a `subsection Function` under `subsection Adiabatic`.
+This function has a expression of `0` so it would not affect the initial temperature
+field but it should have the same number of terms as the `Number of fields` set in the
+`Compositional fields`. Please see [ASPECT forum discussion](https://community.geodynamics.org/t/there-is-a-problem-when-use-initial-temperature-model-adiabatic-half-space-cooling-with-number-of-composition-greater-than-1-ascii-model/3160/2) for more information.
+
+
+The complete input file can be found in
+[cookbooks/multicomponent_steinberger/doc/steinberger_thermochemical_plume.prm](https://www.github.com/geodynamics/aspect/blob/main/cookbooks/multicomponent_steinberger/doc/steinberger_thermochemical_plume.prm).
+
+## Results
+
+We run the model for 300 million years. Plumes from the initial thermochemical
+anomaly as well as from the thermal boundary layer. However, no plumes has risen
+up to the base of the lithosphere due to the mid-mantle viscosity hump ({numref}`fig:viscosity.multicompstn`)
+and the low buoyancy flux of the plume. You may need to adjust the reference
+viscosity profile and/or decrease the percentage of basaltic materials within the
+initial thermochemical anomaly to get similar results from {cite:t}`Dannberg:Sobolev:2015`.
```

Comment:
I changed the excess temperature to 450 K and plume rise up to the base of the lithosphere at ~250 Ma. I think the boundary layer rising because of the dripping slab. I reduced the age of the surface boundary layer to 50 Ma and the basal layer looks more reasonable now.